drivers/regex/mlx5/mlx5.c

```diff
@@ -58,6 +63,116 @@ mlx5_regex_get_ib_device_match(const struct rte_pci_addr *addr)
 		rte_errno = ENOENT;
 	return ibv_match;
 }
+
+static int mlx5_regex_setup_dev(struct mlx5_regex_priv *priv)
+{
+	struct mlx5dv_pd mlx5_pd = {};
+	struct mlx5dv_obj dv_obj;
+	int ret;
+
+	ret = mlx5dv_devx_query_eqn(priv->ctx, 0, &priv->eqn);
+	if (ret || !priv->eqn) {
+		if (!ret)
+			ret = EINVAL;
+		return ret;
+	}
+
+	priv->pd = ibv_alloc_pd(priv->ctx);
+	if (!priv->pd)
+		return ENOMEM;
+	dv_obj.pd.in = priv->pd;
+	dv_obj.pd.out = &mlx5_pd;
+	mlx5dv_init_obj(&dv_obj, MLX5DV_OBJ_PD);
+	priv->pdn = mlx5_pd.pdn;
+
+	priv->uar = mlx5dv_devx_alloc_uar(priv->ctx, 0);
+	if (!priv->uar) {
+		ibv_dealloc_pd(priv->pd);
+		return ENOMEM;
+	}
+	return 0;
+}
+
+static void mlx5_regex_cleanup_dev(struct mlx5_regex_priv *priv)
+{
+	mlx5dv_devx_free_uar(priv->uar);
+	ibv_dealloc_pd(priv->pd);
```

Comment:
You are mixing devx objects and ibv verbs objects.
We need to be sure this will not cause issues in the future.
For example teardown flow on Ctrl-C.
If a mlx5 lib closes before the application, and libmlx5 maybe unaware of the dependencies created by devx commands.
Maybe better double check with yishai.

Reply:
@yuvalavn each object and their sub objects are dependent. here PD and UAR are unrelated so it is fine to create them as both of them take ibv_context as input.
But will check with Yishai if the code doesn't work.